File: unity/Core/Runtime/EE/IMessageBridge.cs
using System;
using System.Threading.Tasks;

namespace EE {
    using MessageHandler = Func<string, string>;

    public interface IMessageBridge {
        bool RegisterHandler(MessageHandler handler, string tag);
        bool DeregisterHandler(MessageHandler handler, string tag);
        string Call(string tag, string message = "");
        Task<string> CallAsync(string tag, string message = "");
    }
}

File: src/ts/src/core/IMessageBridge.ts
export type MessageHandler = (message: string) => void;

export interface IMessageBridge {
    registerHandler(handler: MessageHandler, tag: string): void;
    deregisterHandler(tag: string): void;
    call(tag: string, message?: string): string;
    callAsync(tag: string, message?: string): Promise<string>;
}